### PR TITLE
feat(SDK): support argument for model run

### PR DESCRIPTION
**Pull request by tianweidut**
## Description
the follow-up  changes:
- use @argument decor refactor `handle-args`
- dump argument json for starwhale console when building model package
- add `--help` output for the user custom arguments 

## Modules
- [x] Python-SDK

## Checklist
- [x] run code format and lint check
- [x] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: client/starwhale/api/_impl/argument.py [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+
+# TODO: use a more elegant way to pass extra cli args
+class ExtraCliArgsRegistry:
+    _args = None
+    _lock = threading.Lock()
+
+    @classmethod
+    def set(cls, args: t.List[str]) -> None:
+        with cls._lock:
```
Comment (anda-ren) [inline client/starwhale/api/_impl/argument.py:22]: is there any multiple threading cases? what if a following thread modifies the args

Comment (tianweidut) [reply client/starwhale/api/_impl/argument.py:22]: Not Sure. Only keep `ExtraCliArgsRegistry` thread safe.

File: client/starwhale/api/_impl/argument.py [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+    def _register_wrapper(func: t.Callable) -> t.Any:
+        # TODO: add `--help` for the arguments
+        # TODO: dump parser to json file when model building
+        # TODO: `@handler` decorator function supports @argument decorator
+        parser = get_parser_from_dataclasses(dataclass_types)
+
+        @wraps(func)
+        def _run_wrapper(*args: t.Any, **kw: t.Any) -> t.Any:
+            dataclass_values = init_dataclasses_values(parser, dataclass_types)
+            if "argument" in kw:
```
[the remark this answers is not in the document]
Comment (tianweidut) [reply client/starwhale/api/_impl/argument.py:72]: `sw_argument` may reduce some conflicts, but the current kwargs inject mechanism still meets conflicts. I will add other parameter to custome injected name for `@starwhale.argument`.

[the remark this answers is not in the document]
Comment (tianweidut) [reply client/starwhale/api/_impl/argument.py:72]: updated

**Review by anda-ren — APPROVED** [review]
LGTM

**Pull request merged**